storage/maria/ma_unique.c

```diff
@@ -142,7 +142,7 @@ ha_checksum _ma_unique_hash(MARIA_UNIQUEDEF *def, const uchar *record)
       if (!length || length > tmp_length)
 	length=tmp_length;			/* The whole blob */
     }
-    end= pos+length;
+    end= (length == 0)? pos : pos + length;
```

Comment:
Did UBSAN really flag here? We would have crashed long before here if `pos` was `NULL`.

Reply:
Likely, a false positive in this case. Will revert this case.